File: projects/m1/034-integer-to-ordinal/solutions/alessiacaporaso/integer-to-ordinal.js
// takes an integer as its only parameter and returns a string
// handle the integers between 1 and 12
// return an empty string if the function is called with an argument outside of this range

const userNumeber = parseInt(prompt("Enter a number between 1 and 12"))

function convertNumber(number){
    const ordinalNumber = ["First", "Second", "Third", "Fourth", "Fifth", "Sixth", "Seventh", "Eighth", "Ninth", "Tenth", "Eleventh", "Twelfth"]
    if(number >= 1 && number <= 12){
        const positionNumber = ordinalNumber[number -1]
        // console.log(number, positionNumber)
        return positionNumber
    } else {
        alert("Error! Can't enter an empty string or a number outside of the range between 1 and 12")
    }
}

console.log(`The number entered is: ${userNumeber}. The ordinal number  is: ${convertNumber(userNumeber)}`)

// function convertNumber(number) {
//   let ordinalNumber = "";
//   switch (number) {
//     case 1:
//       ordinalNumber = "first";
//       break;
//     case 2:
//       ordinalNumber = "second";
//       break;
//     case 3:
//       ordinalNumber = "third";
//       break;
//     case 4:
//       ordinalNumber = "fourth";
//       break;
//     case 5:
//       ordinalNumber = "fifth";
//       break;
//     case 6:
//       ordinalNumber = "sixth";
//       break;
//     case 7:
//       ordinalNumber = "seventh";
//       break;
//     case 8:
//       ordinalNumber = "eighth";
//       break;
//     case 9:
//       ordinalNumber = "ninth";
//       break;
//     case 10:
//       ordinalNumber = "tenth";
//       break;
//     case 11:
//       ordinalNumber = "eleventh";
//       break;
//     case 12:
//       ordinalNumber = "twelfth";
//       break;
//     default:
//       ordinalNumber = "Error! Can't enter an empty string or a number outside of the range between 1 and 12";
//       break;
//   }
//   if(number >= 1 && number <= 12){
//     alert(`The number entered is: ${number}. The ordinal number  is: ${ordinalNumber}`)
//   } else {
//     alert(ordinalNumber)
//   }
// }
// convertNumber(1)
// convertNumber(4)
// convertNumber(7)
// convertNumber(12)
// convertNumber(20)

File: projects/m1/035-the-twelve-days-of-christmas/python/integer_to_ordinal.py
def ordinal_number(integer_number):
    if integer_number == 1:
        return "first"
    elif integer_number == 2:
        return "second"
    elif integer_number == 3:
        return "third"
    elif integer_number == 4:
        return "fourth"
    elif integer_number == 5:
        return "fifth"
    elif integer_number == 6:
        return "sixth"
    elif integer_number == 7:
        return "seventh"
    elif integer_number == 8:
        return "eighth"
    elif integer_number == 9:
        return "ninth"
    elif integer_number == 10:
        return "tenth"
    elif integer_number == 11:
        return "eleventh"
    elif integer_number == 12:
        return "twelfth"
    else:
        return ""

if __name__=="__main__":
    for i in range(1,13):
        print("{} --> ".format(i),end="")
        print(ordinal_number(i))